File: python/coloredThreads.py
#!/usr/bin/env python

import time
from threading import Thread
from termcolor import colored
from random import randint

def RecursiveThing(number, color):
	print colored('{} thread, number is {}.'.format(color, number), color)
	time.sleep(randint(0,9))
	
	if number == 0:
		print colored('{} thread complete!'.format(color), color)
	else:
		RecursiveThing(number - 1, color)
	
colors = ['green','blue','yellow','red','cyan']
threads = []

[threads.append(Thread(target=RecursiveThing, args=(10, color))) for color in colors]
[thread.start() for thread in threads]
[thread.join() for thread in threads]

print('All threads complete!')

File: ruby/coloredThreads.rb
#!/usr/bin/env ruby

require 'colorize'

def RecursiveThing(number,color)
	if(number >= 0)
		puts "#{color} thread, number is #{number}.".colorize(color)
		sleep(1+rand(10))
		RecursiveThing(number-1,color)
	else
		puts "#{color} thread complete!".colorize(color)
	end
end

colors = [:green,:blue,:yellow,:red,:cyan]
threads = []

colors.each {|color|threads << Thread.new{RecursiveThing(10,color)}}
threads.each {|thread| thread.join}
puts "All threads complete!"
